### ISSUE TITLE
24-hour timestamps are presented incorrectly, reporting `24:xx` at 12am instead of `00:xx`.

### ISSUE THREAD

**Issue by haydenmc** [mattermost/mattermost-server#14063]
#### Summary
When sending messages recently at 12am, I notice time stamps of `24:xx` when they should be `00:xx`.
![image](https://user-images.githubusercontent.com/5422053/76734544-676c8100-6720-11ea-9ecb-55bc6d9c2cad.png)

#### Steps to reproduce
Version information:
```
Mattermost Version: 5.20.1
Database Schema Version: 5.20.0
Database: postgres
```
Browser:
```
Microsoft Edge
Version 82.0.446.0 (Official build) dev (64-bit)
```

1. Open Mattermost web app
2. In `Account Settings` under `Display`, set `Clock Display` to `24-hour clock`.
3. Send messages at 12am local time

#### Expected behavior
Expect time stamps to be in the format `00:xx` (for example, `00:25` if sent at `12:25 am`).

#### Observed behavior (that appears unintentional) 
Time stamps are in the format `24:xx` (for example, `24:25` if sent at `12:25 am`).

#### Possible fixes
Have not investigated. Doesn't appear to be a browser issue because the incorrect value is a text node in the DOM tree.
![image](https://user-images.githubusercontent.com/5422053/76734929-2d4faf00-6721-11ea-8fb0-6f5ee328e613.png)

**Comment by amyblais** [comment]
There is a ticket open for this here: https://mattermost.atlassian.net/browse/MM-22595.

**Comment by amyblais** [comment]
This is fixed in upcoming v5.22,
